frontend/src/components/edit-profile/edit-profile.tsx

```diff
@@ -0,0 +1,257 @@
+import { ButtonOutline } from '@components/common/button-outline/button-outline';
+import { InputField } from '@components/common/input-field/input-field';
+import { SelectField } from '@components/common/select-field/select-field';
+import { DialogDeleteAccount } from '@components/edit-profile/dialog-delete-account/dialog-delete-account';
+import { SelectYearsRange } from '@components/edit-profile/select-years-range/select-years-range';
+import { SignIn } from '@components/edit-profile/sign-in/sign-in';
+import { useAppForm } from '@hooks/app-form/app-form.hook';
+import { Alert, MenuItem, Stack } from '@mui/material';
+import {
+  ProfileFieldsRequestData,
+  useDeleteUserProfileMutation,
+  useUpdateUserProfileMutation,
+} from '@store/queries/user/update-user';
+
+import styles from './styles.module.scss';
+
+interface EditProfileProps {
+  onClose: () => void;
+}
+
+export const EditProfile: React.FC<EditProfileProps> = ({ onClose }) => {
+  const user = {
+    sex: null,
+    birthYear: '1996',
+    location: 'kyiv',
+    name: 'stepan shevchenko',
+    phone: '+380938889922',
+    email: 'stepan1909@gmail.com',
+  };
+  const navigate = useNavigate();
+  const [openDialog, setOpenDialog] = useState(false);
+  const [
+    updateUserProfile,
+    {
+      isLoading: updateIsLoading,
+      isSuccess: updateIsSuccess,
+      error: updateError,
+    },
+  ] = useUpdateUserProfileMutation();
+
+  const [
+    deleteUserProfile,
+    {
+      isLoading: deleteIsLoading,
+      isSuccess: deleteIsSuccess,
+      error: deleteError,
+    },
+  ] = useDeleteUserProfileMutation();
+
+  const { control, errors, handleSubmit, setValue } =
+    useAppForm<ProfileFieldsRequestData>({
+      defaultValues: {
+        sex: user.sex || 'not_appliable',
+        birthYear: user.birthYear || 'not_appliable',
+        location: user.location || 'not_appliable',
+        name: user.name,
+        phone: user.phone,
+        email: user.email,
+      },
+      validationSchema: updateUserSchema,
+    });
+
+  useEffect(() => {
+    if (deleteIsSuccess) {
+      navigate(AppRoute.ROOT);
+    }
+  });
+
+  const onSubmit: SubmitHandler<ProfileFieldsRequestData> = async (data) => {
+    const updatedUser = await updateUserProfile(data);
+
+    if ('data' in updatedUser) {
+      const { data } = updatedUser;
+
+      for (const [key, value] of Object.entries(data)) {
+        setValue(key, value);
+      }
+    }
+  };
+
+  const handleClickOpenDialog = (): void => {
+    setOpenDialog(true);
+  };
+
+  const handleCloseDialog = (): void => {
+    setOpenDialog(false);
+  };
+
+  const handleDeleteProfile = async (): Promise<void> => {
+    setOpenDialog(false);
+    await deleteUserProfile();
+  };
+
+  return (
+    <div className={styles.popupWrapper}>
+      <DialogDeleteAccount
+        handleCloseDialog={handleCloseDialog}
+        handleDeleteProfile={handleDeleteProfile}
+        isOpen={openDialog}
+      />
+      <div className={styles.popup}>
```

Comment:
This is not the popup, it's the modal window. And for the `Modal` you could use the MUI's `Modal` component.